src/lib/utils.ts

```diff
@@ -34,3 +37,47 @@ export const loadScriptTag = async (srcValue: string): Promise<void> => {
 
   await loadedPromise
 }
+
+export const getUserVariables = (): {userId: string, courseId: string, instance: string} | null => {
+  const userId = '2'
+  const courseId = window.M?.cfg.courseId
+  const instance = window.location.host
+  if (userId === undefined || courseId === undefined || instance === undefined) {
+    return null
+  }
+  return {
+    userId,
+    courseId,
+    instance
+  }
+}
+
+function getVariantMappings(host: string, courseid: string): string {
+  const mappingData = variantMappings as any
+  const maybeDataMapping = mappingData.default[host]?.courses[courseid]
+  if (maybeDataMapping !== undefined) {
+    return maybeDataMapping
+  } else {
+    return 'main'
+  }
+}
+
+export const getVariant = (variants: ContentVariant[]): string => {
+  const userValues = getUserVariables()
+  let varName = 'main'
+  if (userValues != null) {
+    varName = getVariantMappings(userValues.instance, userValues.courseId)
+    console.log('VarName: ' + varName)
+  }
+
+  let html
+  variants.forEach(item => {
+    if (item.variant === varName) {
+      html = item.html
+    }
+  })
+  if (html === undefined) {
+    html = variants[0].html // return whatever variant exists
+  }
+  return html
```

Comment:
In so much as it might be useful for debug purposes to render an error if we unexpectedly find no matching variatn, what do you think about changing this function signature a bit?:

```javascript
export const getVariant = (variants: ContentVariant[]): string | undefined => {
  const userValues = getUserVariables()

  const varName = getVariantMappings(userValues.instance, userValues.courseId)

  const maybeMatch = variants.find(item => item.variant === varName)
  if (maybeMatch === undefined) {
    return maybeMatch
  } else {
    return maybeMatch.html
  }
}
```

With :point_up: , `ContentLoader.tsx` can then display an error:

```javascript
      const data = await response.json() as ContentResponse
      const htmlContent = getVariant(data.content)

      if (htmlContent === undefined) {
        setFetchStatus(FetchStatus.FetchFailure)
        return
      }
```